metadata-ingestion/src/datahub/ingestion/source/metabase.py

```diff
@@ -583,7 +634,9 @@ def get_datasource_from_id(self, datasource_id):
                 key=f"metabase-datasource-{datasource_id}",
                 reason=f"Unable to retrieve Datasource. " f"Reason: {str(http_error)}",
             )
-            return None, None
+            # returning empty string as `platform` because
+            # `make_dataset_urn_with_platform_instance()` only accepts `str`
+            return "", None, None, None
```

Comment:
If platform is "", we should not even call make_dataset_urn_with_platform_instance, right ?

Reply:
I've added the check for empty string in `get_datasource_urn()` for fast return of `None`. Could you please have a look?

Reply:
can also return None if check is changed to "if not platform"

Reply:
Returning `None` as first element of tuple from `get_datasource_from_id()` will cause type hints verification to fail as `make_dataset_urn_with_platform_instance()` only accepts `str` and I'd rather not change common method in scope of this PR.